### PR TITLE
[json-file-plus] Improve types

### PR DESCRIPTION
**Pull request by BendingBender**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/ljharb/json-file-plus
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/json-file-plus/index.d.ts [hunk 1]
```diff
@@ -2,36 +2,118 @@
...
+     */
+    function sync(filename: string): JSONFile;
 
-  interface Format {
-    readonly indent: number | '\t';
-    readonly trailing: boolean;
-  }
+    interface Format {
+        indent: number | '\t' | ' ';
+        trailing: boolean;
```
Comment (ljharb) [inline types/json-file-plus/index.d.ts:69]: why are these no longer readonly?

File: types/json-file-plus/index.d.ts [hunk 2]
```diff
@@ -2,36 +2,118 @@
...
+         */
+        data: any;
+        constructor(raw: string);
+        /**
+         * Get top-level keys.
+         *
+         * @param key The key to get. If empty, the entire data will be returned.
+         */
+        get(key: PropertyKey, callback?: (err: Error | null, value: any) => void): Promise<any>;
+        get(callback?: (err: Error | null, value: any) => void): Promise<any>;
```
Comment (ljharb) [inline types/json-file-plus/index.d.ts:88]:
```suggestion
        get(callback?: (err: Error | null, value: unknown) => void): Promise<unknown>;
```

File: types/json-file-plus/index.d.ts [hunk 3]
```diff
@@ -2,36 +2,118 @@
...
+         * Direct access to the data from the file.
+         */
+        data: any;
+        constructor(raw: string);
+        /**
+         * Get top-level keys.
+         *
+         * @param key The key to get. If empty, the entire data will be returned.
+         */
+        get(key: PropertyKey, callback?: (err: Error | null, value: any) => void): Promise<any>;
```
Comment (ljharb) [inline types/json-file-plus/index.d.ts:87]:
```suggestion
        get(key: PropertyKey, callback?: (err: Error | null, value: unknown) => void): Promise<unknown>;
```

**Review by ljharb — COMMENTED** [review]
LGTM overall!

File: types/json-file-plus/index.d.ts (hunk 1 above)
Comment (BendingBender) [reply types/json-file-plus/index.d.ts:69]: Because they really aren't. And the readme explicitly states: "extracted formatting data. change at will."

Comment (ljharb) [reply types/json-file-plus/index.d.ts:71]: Fair enough.

**Review by ljharb — APPROVED** [review]

**Comment by ljharb** [comment]
Ready to merge

**Pull request merged**